wear/src/main/java/io/homeassistant/companion/android/sensors/HealthServicesSensorManager.kt

```diff
@@ -90,21 +141,61 @@ class HealthServicesSensorManager : SensorManager {
                         UserActivityState.USER_ACTIVITY_EXERCISE -> "exercise"
                         else -> "unknown"
                     },
-                    when (info.userActivityState) {
-                        UserActivityState.USER_ACTIVITY_EXERCISE -> "mdi:run"
-                        UserActivityState.USER_ACTIVITY_ASLEEP -> "mdi:sleep"
-                        UserActivityState.USER_ACTIVITY_PASSIVE -> "mdi:human-handsdown"
-                        else -> userActivityState.statelessIcon
-                    },
+                    getActivityIcon(info),
                     mapOf(
                         "time" to info.stateChangeTime,
                         "exercise_type" to info.exerciseInfo?.exerciseType?.name
-                    )
+                    ),
+                    forceUpdate = info.userActivityState == UserActivityState.USER_ACTIVITY_EXERCISE
                 )
 
                 SensorWorker.start(latestContext)
             }
 
+            override fun onNewDataPointsReceived(dataPoints: DataPointContainer) {
+                Log.d(TAG, "New data point received: ${dataPoints.dataTypes}")
+                val floorsDaily = dataPoints.getData(DataType.FLOORS_DAILY)
+                val bootInstant =
+                    Instant.ofEpochMilli(System.currentTimeMillis() - SystemClock.elapsedRealtime())
+                var latest = 0
+                var lastIndex = 0
+
+                dataPoints.dataTypes.forEachIndexed { _, dataType ->
+
+                    if (dataType is DeltaDataType<*, *>) {
+                        val data = dataPoints.getData(dataType)
+                        data.forEachIndexed { indexPoint, dataPoint ->
+                            if (dataPoint is IntervalDataPoint) {
+                                val endTime = dataPoint.getEndInstant(bootInstant)
+                                Log.d(
+                                    TAG,
+                                    "Data for ${dataType.name} index: $indexPoint with value: ${dataPoint.value} end time: ${endTime.toEpochMilli()}"
+                                )
+                            }
+                        }
+                    }
+                }

```

Comment:
This doesn't appear to get used, is this just for logging or is there another reason for doing this? (The emulator doesn't create data so I can't test this, but the daily floors specific loop below seems to do almost the same and that result is used.)

Reply:
Yup just for logging to help out when adding additional sensors from health services.